### PR TITLE
Html css popup

### PR DESCRIPTION
**Pull request by nadsatt**
### html-css-popup

[Demo](https://nadsatt.github.io/frontend-2021-homeworks/submissions/nadsatt/html-css-popup) |
[Code base](https://github.com/nadsatt/frontend-2021-homeworks/tree/html-css-popup/submissions/nadsatt/html-css-popup)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Review by MaxKalashnyk — CHANGES_REQUESTED** [review]
Great job! Please check your preview in the [validator](https://validator.w3.org/)

File: submissions/nadsatt/html-css-popup/style.css [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+* {
+	vertical-align: baseline;
+	font-weight: inherit;
+	font-family: inherit;
+	font-style: inherit;
+	font-size: 100%;
+	border: 0 none;
+	outline: 0;
+	padding: 0;
+    margin: 0;
```
Comment (natashafir) [inline submissions/nadsatt/html-css-popup/style.css:34]: Looks great, just need сode formatting here.

**Review by natashafir — CHANGES_REQUESTED** [review]

File: submissions/nadsatt/html-css-popup/style.css (hunk 1 above)
Comment (nadsatt) [reply submissions/nadsatt/html-css-popup/style.css:34]: Thanks for review! Defined issues fixed.

**Comment by ses3332002** [comment]
"Contacts" icon is 25*25 px while other icons are 30*30 px, so caption "Contacts" is higher then other captions.

**Comment by ses3332002** [comment]
I really liked your design, except two scroll bars in the drop-down menu (when it has not yet been "expanded" with additional items). I viewed in Chrome and Firefox on Windows.

**Review by MaxKalashnyk — APPROVED** [review]
Awesome work! I appreciate the design of your app.

File: submissions/nadsatt/html-css-popup/style.css [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+.link__avatar {
+    border: 2px solid var(--color-gray-light);
+    border-radius: 50%;
+}
+
+.list-item__link .link__icon {
+    fill: var(--color-gray-light);
+}
+
+.list-item__checkbox {
```
Comment (A-Ostrovnyy) [inline submissions/nadsatt/html-css-popup/style.css:95]: Your approach of hiding checkbox good but only for one situation:
Watch [video](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s) how to hide checkbox in the right way

File: submissions/nadsatt/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,294 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>popup</title>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <link rel="stylesheet" href="style.css">
+        helllo
```
Comment (A-Ostrovnyy) [inline submissions/nadsatt/html-css-popup/index.html]: Hello)

File: submissions/nadsatt/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                </li>
+                <li class="list-item">
+                    <input class="list-item__checkbox" type="checkbox">
+                    <a class="list-item__link">
+                        <svg class="link__icon" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="24" height="24" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve">
+                            <path d="M186.2,139.6h139.6V0H186.2V139.6z M372.4,0v139.6H512V0H372.4z M0,139.6h139.6V0H0V139.6z M186.2,325.8h139.6V186.2H186.2 V325.8z M372.4,325.8H512V186.2H372.4V325.8z M0,325.8h139.6V186.2H0V325.8z M186.2,512h139.6V372.4H186.2V512z M372.4,512H512 V372.4H372.4V512z M0,512h139.6V372.4H0V512z"/>
+                        </svg>
+                    </a>
+                    <div class="list-item__popup">
+                       <div class="popup__items--visible">
```
Comment (A-Ostrovnyy) [inline submissions/nadsatt/html-css-popup/index.html:26]: If you use BEM methodology use class-modificator for the state of elements. Also, class-modificator can't be without this  class no modificator.
[Read more about modificator](https://ru.bem.info/methodology/quick-start/#%D0%9C%D0%BE%D0%B4%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%82%D0%BE%D1%80)
BEM element can't exist outside his BEM block. I can't see block `.popup` of element `.popup__items`.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
See my comments above!

**Comment by nadsatt** [comment]
@A-Ostrovnyy all ur comments taken into account. Please re-review

File: submissions/nadsatt/html-css-popup/index.html [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+                    <input class="list__check-input check-input" id="list-check-input" type="checkbox">
+                    <label class="list__check-label list__link" for="list-check-input">
+                       <svg class="list__check-icon list__icon" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="24" height="24" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve">
+                            <path d="M186.2,139.6h139.6V0H186.2V139.6z M372.4,0v139.6H512V0H372.4z M0,139.6h139.6V0H0V139.6z M186.2,325.8h139.6V186.2H186.2 V325.8z M372.4,325.8H512V186.2H372.4V325.8z M0,325.8h139.6V186.2H0V325.8z M186.2,512h139.6V372.4H186.2V512z M372.4,512H512 V372.4H372.4V512z M0,512h139.6V372.4H0V512z"/>
+                        </svg>
+                    </label>
+ 
+                    <div class="list__popup">
+                        <div class="popup__top-container">
```
Comment (A-Ostrovnyy) [inline submissions/nadsatt/html-css-popup/index.html:28]: In this case, we have BEM element `popup__top-container`, but I can't see a parent block `popup`.
Please take a look at this issue.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good job.
One more issue)

File: submissions/nadsatt/html-css-popup/index.html (hunk 5 above)
Comment (nadsatt) [reply submissions/nadsatt/html-css-popup/index.html:28]: Done

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**